src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -858,35 +849,28 @@ private RuntimeFieldInfo[] PopulateFields(Filter filter)
 
                 private unsafe void PopulateRtFields(Filter filter, RuntimeType declaringType, ref ListBuilder<RuntimeFieldInfo> list)
                 {
-                    IntPtr* pResult = stackalloc IntPtr[64];
-                    int count = 64;
-
-                    if (!RuntimeTypeHandle.GetFields(declaringType, pResult, &count))
+                    Span<IntPtr> result = stackalloc IntPtr[64];
+                    int count;
+                    while (!RuntimeTypeHandle.GetFields(declaringType, result, out count))
                     {
-                        fixed (IntPtr* pBigResult = new IntPtr[count])
-                        {
-                            RuntimeTypeHandle.GetFields(declaringType, pBigResult, &count);
-                            PopulateRtFields(filter, pBigResult, count, declaringType, ref list);
-                        }
-                    }
-                    else if (count > 0)
-                    {
-                        PopulateRtFields(filter, pResult, count, declaringType, ref list);
+                        Debug.Assert(count > result.Length);
+                        result = new IntPtr[count];
                     }
+                    PopulateRtFields(filter, result.Slice(0, count), declaringType, ref list);
```

Comment:
```suggestion
                    if (count > 0)
                    {
                        PopulateRtFields(filter, result.Slice(0, count), declaringType, ref list);
                    }
```
Avoid some unnecessary code for types without fields like the original code.